docs/content/latest/architecture/docdb-sharding/tablet-splitting.md

```diff
@@ -195,21 +206,94 @@ Tablet UUID                       Range                                 Leader
+
+1. Create a three-node cluster.
+
+```sh
+./bin/yb-ctl --rf=3 create --num_shards_per_tserver=1 --ysql_num_shards_per_tserver=1 --master_flags '"tablet_split_size_threshold=30000000"' --tserver_flags '"memstore_size_mb=10"'
+```
+
+2. Create a table for workload.
+
+```sh
+./bin/ysqlsh -c "CREATE DATABASE ycsb;"
+./bin/ysqlsh -d ycsb -c "CREATE TABLE usertable (
+           YCSB_KEY TEXT,
+           FIELD0 TEXT, FIELD1 TEXT, FIELD2 TEXT, FIELD3 TEXT,
+           FIELD4 TEXT, FIELD5 TEXT, FIELD6 TEXT, FIELD7 TEXT,
+           FIELD8 TEXT, FIELD9 TEXT,
+           PRIMARY KEY (YCSB_KEY ASC));"
+```
+
+3. Create the properties file for YCSB at `~/code/YCSB/db-local.properties` and add the following content.
+
+```sh
+db.driver=org.postgresql.Driver
+db.url=jdbc:postgresql://127.0.0.1:5433/ycsb;jdbc:postgresql://127.0.0.2:5433/ycsb;jdbc:postgresql://127.0.0.3:5433/ycsb
+db.user=yugabyte
+db.passwd=
+core_workload_insertion_retry_limit=10
+```
+
+4. Start loading data for the workload.
+
+```sh
+~/code/YCSB/bin/ycsb load jdbc -s -P ~/code/YCSB/db-local.properties -P ~/code/YCSB/workloads/workloada -p recordcount=500000 -p operationcount=1000000 -p threadcount=4
+```
+
+5. Monitor the tablets splitting by going to the YugabyteDB Web UI at `http://localhost:7000/tablet-servers` and `http://127.0.0.1:9000/tablets`.
+
+6. Run the workload.
+
+```sh
+~/code/YCSB/bin/ycsb run jdbc -s -P ~/code/YCSB/db-local.properties -P ~/code/YCSB/workloads/workloada -p recordcount=500000 -p operationcount=1000000 -p threadcount=4
+```
+
+7. Get the list of tablets accessed during the run phase of the workload.
+
+```sh
+diff -C1 after-load.json after-run.json | grep tablet_id | sort | uniq
+```
+
+8. The list of tablets accessed can be compared with `http://127.0.0.1:9000/tablets` to make sure no presplit tablets have been accessed.
+
+## Current tablet splitting limitations
+
+Presplitting tablets and automatic tablet splitting are in beta. To follow the work-in-progress on tablet splitting, see [GitHub #1004](https://github.com/yugabyte/yugabyte-db/issues/1004).
+
+Here are known limitations that are planned to be resolved in the next releases:
 
+* Presplit tablets remain in the system forever and are not deleted from the disk.
+* There is no upper bound on the number of tablets for the table when automatic tablet splitting is enabled.
+* During tablet splitting, client applications can get an error from the driver and need to retry the request.
+* If tablet splitting occurs during an ongoing distributed transaction, it could be aborted and need to be retried.
+* Because splitting of tablets that are not completely compacted is not yet implemented, tablets created by tablet splitting might be split after they reach the specified size threshold.
```

Comment:
since we already mention that these will be resolved eventually, we can reword this to something like

`Tablets created as a result of splitting cannot be split again, until they are fully compacted. This could mean that the split will happen much later than after the hit the user specified size threshold.`